comctl32.go

```diff
@@ -388,3 +394,191 @@ func LoadIconWithScaleDown(hInstance HINSTANCE, lpIconName *uint16, w int32, h i
+func (u *TASKDIALOG_BUTTON_UNPACKED) Pack(buf *bytes.Buffer) error {
+	if err := binary.Write(buf, packByteOrder, u.ButtonID); err != nil {
+		return err
+	}
+	return binary.Write(buf, packByteOrder, packPtr(uintptr(unsafe.Pointer(u.ButtonText))))
+}
+
+type TASKDIALOG_ELEMENTS int32
+
+const (
+	TDE_CONTENT              = TASKDIALOG_ELEMENTS(0)
+	TDE_EXPANDED_INFORMATION = TASKDIALOG_ELEMENTS(1)
+	TDE_FOOTER               = TASKDIALOG_ELEMENTS(2)
+	TDE_MAIN_INSTRUCTION     = TASKDIALOG_ELEMENTS(3)
+)
+
+type TASKDIALOG_ICON_ELEMENTS int32
+
+const (
+	TDIE_ICON_MAIN   = TASKDIALOG_ICON_ELEMENTS(0)
+	TDIE_ICON_FOOTER = TASKDIALOG_ICON_ELEMENTS(1)
+)
+
+const (
+	TD_WARNING_ICON     = uintptr(0xFFFF)
+	TD_ERROR_ICON       = uintptr(0xFFFE)
+	TD_INFORMATION_ICON = uintptr(0xFFFD)
+	TD_SHIELD_ICON      = uintptr(0xFFFC)
+)
+
+type TASKDIALOG_COMMON_BUTTON_FLAGS uint32
+
+const (
+	TDCBF_OK_BUTTON     = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0001)
+	TDCBF_YES_BUTTON    = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0002)
+	TDCBF_NO_BUTTON     = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0004)
+	TDCBF_CANCEL_BUTTON = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0008)
+	TDCBF_RETRY_BUTTON  = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0010)
+	TDCBF_CLOSE_BUTTON  = TASKDIALOG_COMMON_BUTTON_FLAGS(0x0020)
+)
+
+// TASKDIALOGCONFIG_UNPACKED is nearly identical to TASKDIALOGCONFIG in the Windows
+// SDK, except for the fact that that the SDK version is packed. Since Go cannot
+// grok that, we implement a Pack method on this struct to encode its contents
+// correctly for Windows.
+type TASKDIALOGCONFIG_UNPACKED struct {
+	Size                 uint32 // Size is included for consistency with the SDK but is ignored by Pack()
+	HWNDParent           HWND
+	HInstance            HINSTANCE
+	Flags                TASKDIALOG_FLAGS
+	CommonButtons        TASKDIALOG_COMMON_BUTTON_FLAGS
+	WindowTitle          *uint16
+	MainIcon             uintptr
+	MainInstruction      *uint16
+	Content              *uint16
+	CButtons             uint32
+	PButtons             *TASKDIALOG_BUTTON
+	DefaultButton        int32
+	CRadioButtons        uint32
+	PRadioButtons        *TASKDIALOG_BUTTON
+	DefaultRadioButton   int32
+	VerificationText     *uint16
+	ExpandedInformation  *uint16
+	ExpandedControlText  *uint16
+	CollapsedControlText *uint16
+	FooterIcon           uintptr
+	Footer               *uint16
+	Callback             uintptr
+	CallbackData         uintptr
+	Width                uint32
+}
+
+// TASKDIALOGCONFIG is opaque because it is a packed data structure.
+// Use (*TASKDIALOGCONFIG_UNPACKED).Pack() to obtain one.
+type TASKDIALOGCONFIG struct {
+	_ [sizeTASKDIALOGCONFIG]byte
+}
+
+// Pack takes the contents of u and writes them in a packed format to a
+// TASKDIALOGCONFIG, which is returned as the result.
+func (u *TASKDIALOGCONFIG_UNPACKED) Pack() *TASKDIALOGCONFIG {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeTASKDIALOGCONFIG))
+
+	v := reflect.ValueOf(u).Elem()
```

Comment:
Would it be better to just explicitly do each field instead of using `reflect`?

Reply:
I'm going to leave this, as I expect that I would create more copypasta errors that way :-)